core/src/main/java/bisq/core/dao/state/DaoStateService.java

```diff
@@ -646,6 +646,10 @@ public boolean isLockupOutput(TxOutput txOutput) {
                 .findFirst());
     }
 
+    public Optional<TxOutput> getLockupOpReturnTxOutput(String txId) {
```

Comment:
Perhaps a check to make there is actually an opreturn output would be good

Reply:
like: 
  public Optional<TxOutput> getLockupOpReturnTxOutput(String txId) {
        return getTx(txId).map(Tx::getLastTxOutput).filter(txOutput -> txOutput.getOpReturnData() != null);
    }

??

Reply:
Yes, perfect